shell/platform/embedder/tests/embedder_test_context.cc

```diff
@@ -16,16 +17,32 @@ EmbedderTestContext::EmbedderTestContext(std::string assets_path)
       native_resolver_(std::make_shared<TestDartNativeResolver>()) {
   auto assets_dir = fml::OpenDirectory(assets_path_.c_str(), false,
                                        fml::FilePermission::kRead);
-  vm_snapshot_data_ =
-      fml::FileMapping::CreateReadOnly(assets_dir, "vm_snapshot_data");
-  isolate_snapshot_data_ =
-      fml::FileMapping::CreateReadOnly(assets_dir, "isolate_snapshot_data");
 
   if (flutter::DartVM::IsRunningPrecompiledCode()) {
-    vm_snapshot_instructions_ =
-        fml::FileMapping::CreateReadExecute(assets_dir, "vm_snapshot_instr");
-    isolate_snapshot_instructions_ = fml::FileMapping::CreateReadExecute(
-        assets_dir, "isolate_snapshot_instr");
+    std::string filename(assets_path_);
+    filename += "/app_elf_snapshot.so";
+
+    const uint8_t *vm_snapshot_data = nullptr,
+                  *vm_snapshot_instructions = nullptr,
+                  *isolate_snapshot_data = nullptr,
+                  *isolate_snapshot_instructions = nullptr;
+    const char* error = nullptr;
+
+    elf_library_handle_ = Dart_LoadELF(
```

Comment:
`FML_CHECK(elf_library_handle_ != nullptr)` to just make this a fatal error. This is a test and it makes no sense to keep going past this point.

Also, please free the `error` (even if you move to the assertion) since folks use may use this to understand a pattern while reading this bit of code.

Reply:
Done, although the error returned by Dart_LoadELF is not supposed to be freed (it's just one of a set of static strings).